src/suggestions.js

```diff
@@ -0,0 +1,83 @@
+import {computeAccessibleName} from 'dom-accessibility-api'
+import {getRoles} from './role-helpers'
+import {getDefaultNormalizer} from './matches'
+
+const normalize = getDefaultNormalizer()
+
+function getLabelTextFor(element) {
+  let label
+
+  const allLabels = Array.from(document.querySelectorAll('label'))
+
+  label = allLabels.find(lbl => lbl.control === element)
+
+  if (!label) {
+    const ariaLabelledBy = element.getAttribute('aria-labelledby')
+
+    if (ariaLabelledBy) {
+      // we're using this notation because with the # selector we would have to escape special characters e.g. user.name
+      // see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Escaping_special_characters
+      label = document.querySelector(`[id=${ariaLabelledBy}]`)
+    }
+  }
+
+  if (label) {
+    return label.textContent
+  }
+
+  return undefined
+}
+
+function makeSuggestion(queryName, content, name) {
+  return {
+    queryName,
+    toString() {
+      const options = name ? `, {name:/${name}/}` : ''
```

Comment:
```suggestion
      const options = name ? `, {name: /${name.toLowerCase()}/i}` : ''
```

Reply:
@kentcdodds was talking about that with @appleJax (we were pairing on zoom for this - and you should totally @allcontributors him 😄)

wouldn't it be better to be exact on this?  like, you have the exact text of the element?  i'm on the fence would like to hear your thoughts though...

Reply:
Why would you want your test to break, due to simple changes in capitalization?

Reply:
@smeijer I'm kinda on the fence, I see both arguments.  On the one hand I agree that maybe a change in capitalization shouldn't break the test, on the other, you could argue that a change in capitalization _is_ a change in content and it should break the test.  But more specifically in this case we have the actual content of the element in hand and we can give an exact value that will not break the test.

Reply:
> But more specifically in this case we have the actual content of the element in hand and we can give an exact value that will not break the test.

That part is for future changes of course 😁 now it will work. Than one day some one will come, and decides that the labels should all be capitalized.

This won't affect how the inputs work, or how your form will be submitted. It's a visual change only. We don't want tests to break in visual changes. For that, other test methods exist.

Reply:
agree to disagree, I think some folks would call it a content change, something that the user sees.

Reply:
I see the capitalization like styling. It doesn't matter if the button is blue or green, I'm gonna click it if it says: "Win a million dollars" 😉

I won't care if that's in all caps or all lower case either: "WIN A MILLION DOLLARS" or "win a million DOlLaRs". I'm clicking that either way 😜

Reply:
haha. You had me at win a million dollars. Hold off on merging I'm gonna push the changes you suggested...

Reply:
all set ready to go